pretyping/detyping.ml

```diff
@@ -694,6 +694,13 @@ and detype_binder (lax,isgoal as flags) bk avoid env sigma na body ty c =
       let t = if s != InProp  && not !Flags.raw_print then None else Some (detype (lax,false) avoid env sigma ty) in
       GLetIn (na', c, t, r)
 
+and detype_array flags avoid env sigma args =
+  let ans = ref [] in
+  for i = Array.length args - 1 downto 0 do
+    ans := detype flags avoid env sigma args.(i) :: !ans;
+  done;
+  !ans
```

Comment:
My mistake, you actually mentioned it in the commit message. If the evaluation order matters that much, we could always reuse the code of `CList.map`, so that the evaluation happens left to right.